plugins/ingest/src/main/java/org/elasticsearch/ingest/Pipeline.java

```diff
@@ -65,7 +65,7 @@ public Builder(String name) {
             this.name = name;
         }
 
-        public Builder(Map<String, Object> config) {
+        public Builder(Map<String, Object> config, Map<String, Processor.Builder> processorRegistry) {
```

Comment:
There's a bit of a problem here. A builder as it is presented here, is a stateful construct, meaning, you use a builder to build a "configured" instance of a processor. But the passed in registry should not be stateful... effectively, the builders in the passed in registry should act as "factories". 

either separate the two notions (builder & factory) or have the registry hold "prototype" builders and then change the `void fromMap(Map)` to `Processor build(Map)`. Personally I like the separation of the constructs - builds clear API with clear roles and responsibility for each class. But core moved more towards the "prototype" direction, so perhaps a prototype here will be more aligned with core.